ocrd_webapi/authentication.py

```diff
@@ -0,0 +1,52 @@
+from hashlib import sha512
+from random import random
+from typing import Tuple
+
+from .database import create_user, get_user
+
+
+async def authenticate_user(email: str, password: str):
+    db_user = await get_user(email=email)
+    if not db_user:
+        raise ValueError(f"User not found: {email}")
+    validation_status = validate_password(
+        plain_password=password,
+        encrypted_password=db_user.encrypted_pass
+    )
+    if not validation_status:
+        raise ValueError(f"User was not validated: {email}")
+
+
+async def register_user(email: str, password: str):
+    salt, encrypted_password = encrypt_password(password)
+    db_user = await get_user(email)
+    if db_user:
+        raise ValueError(f"User is already registered: {email}")
+    created_user = await create_user(
+        email=email,
+        encrypted_pass=encrypted_password,
+        salt=salt,
+        validated_account=False
+    )
+    if not created_user:
+        raise ValueError(f"Failed to register user: {email}")
+
+
+def encrypt_password(plain_password: str) -> Tuple[str, str]:
+    salt = get_random_salt()
+    hashed_password = get_hex_digest(salt, plain_password)
+    encrypted_password = f'{salt}${hashed_password}'
+    return salt, encrypted_password
+
+
+def get_hex_digest(salt: str, plain_password: str):
+    return sha512(f'{salt}{plain_password}'.encode('utf-8')).hexdigest()
+
+
+def get_random_salt() -> str:
+    return sha512(f'{str(random())}{hash}'.encode('utf-8')).hexdigest()[:8]
+
+
+def validate_password(plain_password: str, encrypted_password: str) -> bool:
+    salt, hashed_password = encrypted_password.split('$')
```

Comment:
```suggestion
    salt, hashed_password = encrypted_password.split('$', 1)
```

Password should be `0-9a-f` since it's hex digest, so should not contain `$` but just in case it might in the future.